DARPA-ASKEM/TERArium
# chore: start reorganizing models/petrinets

### Summary
There are a couple of things in this PR, the main gist is to consolidate scattered and (arguably) misplaced code fragment under the same roof - we have been sort of intermingling "petrinet" and "model" both conceptually and within code logic. As a precursor to more advanced model-editor work, this PR addresses these inconsistencies:

1. Extract out the skeletons of a de-facto PetriNet editor that we will reuse across the application. Current each page has it's own renderer to do the job

2. Consolidate types/services. We have a blurry line of what a Model is and what a PetriNet is, likewise between types/utilities/services. In this PR we try to sort this out:
   - Create a src/petrinet folder to hold all things specifically related to petrinet
   - Created a "ConcreateModel<type>" to accommodate various model-frameworks. The current "Model" is aliased as "ConcreateType<any>".  This makes it a little easier to _not_ mishmash petrinet and model together...though subject to change pending exactly how we'd adopt new frameworks down the road.

### Testing
There should be no functional changes. App works as before:
- Check model page
- Check the experimental model-editor page
